src/main/java/org/openrewrite/java/spring/framework/MigrateWebMvcConfigurerAdapter.java

```diff
@@ -59,18 +65,63 @@ public J.ClassDeclaration visitClassDeclaration(J.ClassDeclaration classDecl, Ex
                     }
                     cd = JavaTemplate.builder("WebMvcConfigurer")
                             .contextSensitive()
-                            .imports("org.springframework.web.servlet.config.annotation.WebMvcConfigurer")
+                            .imports(WEB_MVC_CONFIGURER)
                             .javaParser(JavaParser.fromJavaVersion()
-                                    .classpathFromResources(ctx, "spring-webmvc-5.*"))
+                                    .classpathFromResources(ctx, "spring-webmvc-5"))
                             .build().apply(getCursor(), cd.getCoordinates().addImplementsClause());
                     updateCursor(cd);
                     cd = (J.ClassDeclaration) new RemoveSuperStatementVisitor().visitNonNull(cd, ctx, getCursor().getParentOrThrow());
-                    maybeRemoveImport("org.springframework.web.servlet.config.annotation.WebMvcConfigurerAdapter");
-                    maybeAddImport("org.springframework.web.servlet.config.annotation.WebMvcConfigurer");
+                    maybeRemoveImport(WEB_MVC_CONFIGURER_ADAPTER);
+                    maybeAddImport(WEB_MVC_CONFIGURER);
                 }
                 return cd;
             }
 
+            @Override
+            public J.NewClass visitNewClass(J.NewClass newClass, ExecutionContext ctx) {
+                if (newClass.getClazz() != null && TypeUtils.isOfClassType(newClass.getClazz().getType(), WEB_MVC_CONFIGURER_ADAPTER)) {
+                    if (newClass.getClazz() instanceof J.Identifier) {
+                        J.Identifier identifier = (J.Identifier) newClass.getClazz();
+                        newClass = newClass.withClazz(identifier
+                                .withType(WEB_MVC_CONFIGURER_TYPE)
+                                .withSimpleName(((JavaType.ShallowClass) WEB_MVC_CONFIGURER_TYPE).getClassName())
+                        );
+                    } else if (newClass.getClazz() instanceof J.ParameterizedType) {
+                        J.ParameterizedType parameterizedType = (J.ParameterizedType) newClass.getClazz();
+                        newClass = newClass.withClazz(parameterizedType
+                                .withType(WEB_MVC_CONFIGURER_TYPE)
+                                .withClazz(TypeTree.build(WEB_MVC_CONFIGURER).withType(WEB_MVC_CONFIGURER_TYPE))
+                        );
```

Comment:
Neither `WebMvcConfigurerAdapter` nor `WebMvcConfigurer` takes in an generic type right? Would we even need the handling here then?
```suggestion
```